gitpod-io/gitpod
# Add a new Workspaces list view using react query

## Description

This adds a new version of the Workspaces list view that's using react-query. I opted to build this alongside the existing `<Workspaces/>` UI and put it behind the feature flag `useNewWorkspacesList`.

I do apologize for the size of this PR, there wasn't a clear way to re-use some of the existing UI components that were coupled to things like the `WorkspacesModel`, so I ended up with new versions of a few things. I also took the opportunity to break things down into some more fine-grained components that the new view utilizes.

## Related Issue(s)

Fixes #15673 

## How to test

The feature flag `useNewWorkspacesList` must be enabled for you to see the new Workspaces list UI. This should be enabled by default in the connected preview environment.

* Load the `/workspaces` view.
* Create a new workspace, ensure it shows up in the list afterwards.
* Try all of the different actions that show up in the overflow menu as well.

## Release Notes

```release-note
NONE
```

## Documentation


## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
